### PR TITLE
Import Googletest 1.10.0

### PR DESCRIPTION
**Pull request by emaxx-google**
Add sources of the Googletest project into //third_party/.

This new dependency will be later used in Emscripten builds (as
tracked by #177) in order to run C++ unit tests from our project.
This wasn't needed before, since Googletest was pulled in via
Native Client SDK's webports, but for Emscripten there's no
built-in way to get Googletest.

### CODE REVIEW

**Comment by emaxx-google** [comment]
Note for reviewers: Only the `README.google` file needs to be reviewed; also I'm curious whether I'm not missing some built-in way in Emscripten to pull in Googletest. All other files in this PR are a verbatim copy from the upstream's Googletest repository.

**Review by Fabian-Sommer — APPROVED** [review]
LGTM

File: third_party/googletest/README.google [hunk 1]
```diff
@@ -0,0 +1,7 @@
+URL: https://github.com/google/googletest/archive/release-1.10.0.tar.gz
+Version: 1.10.0
+License: BSD
+License File: LICENSE
+Description:
+Google's C++ test framework.
+Local Modifications:
```
Comment (Fabian-Sommer) [inline third_party/googletest/README.google:7]: nit: either remove this line or add 'None'

**Review by MahmoudAGawad — APPROVED** [review]
LGTM.

I came to know[1] that there is no built-in support for gtest. However, there is this a suggested library[2] which downloads/builds/runs gtests on various platform including emscripten*.

(*) it builds gtest 1.7 though.

[1] https://groups.google.com/g/emscripten-discuss/c/WEYYbA29E6Y
[2] https://github.com/arielm/chronotext-gtest

**Comment by emaxx-google** [comment]
> I came to know[1] that there is no built-in support for gtest. However, there is this a suggested library[2] which downloads/builds/runs gtests on various platform including emscripten*.
> 
> (*) it builds gtest 1.7 though.
> 
> [1] https://groups.google.com/g/emscripten-discuss/c/WEYYbA29E6Y
> [2] https://github.com/arielm/chronotext-gtest

Thanks, I also noticed that existing repository, but decided against using it - I don't know where it's coming from, and also, as you said, it's become pretty outdated now.
Luckily, Googletest's upstream code seems to work under Emscripten as-is, according to my tests.

**Comment by emaxx-google** [comment]
Thanks for quick reviews!

**Pull request merged**